MaxText/layers/attentions.py

```
@@ -710,7 +711,10 @@ def tpu_flash_attention(
     )
 
     mask_shape = (self.config.max_target_length, self.config.max_target_length)
-    mask = splash_attention_mask.CausalMask(shape=mask_shape)
+    if self.attention_type == AttentionType.FULL:
```

Does this mean future flash attention mask will be generated here in the branch?

---

We have been using this branch to generate mask for flash attention. Before this change, we by default started with a causal mask, and add modifications such as chunks on top of it. After this change, we started with a full mask when attention type is full, and likely won't add more modifications due to the conditions for modifications will be False